pallets/foreign-investments/src/impls/mod.rs

```diff
@@ -186,20 +213,22 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 	fn collect_foreign_investment(
 		who: &T::AccountId,
 		investment_id: T::InvestmentId,
-		foreign_payout_currency: T::CurrencyId,
-		pool_currency: T::CurrencyId,
-	) -> Result<ExecutedForeignCollectInvest<T::Balance>, DispatchError> {
+		foreign_payment_currency: T::CurrencyId,
+	) -> DispatchResult {
+		let payment_currency = InvestmentPaymentCurrency::<T>::get(who, investment_id)?;
```

Comment:
Just wandering regarding the states of this. Wouldn't it be possible for the initial `IncreaseInvestOrder` to be fully "done" before a `CollectInvest` is actually triggered? 

Meaning, the `InvestmentPaymentCurrency` state would be `None`.

Reply:
An `InvestmentState` is only cleared if
1. either the entire unprocessed amount was decreased and that swap is fulfilled (`SwapIntoForeignDone` --> `NoState`)
2. The state only holds an unprocessed investment amount which was fully processed (`InvestmentOngoing` --> `NoState`)

The first one is irrelevant here because it requires the `InvestmentState` to be re-initialized by increasing the investment which sets the payment currency.

The second one is only known if the investment is collected. Therefore, we trigger the foreign collect flow before clearing the `InvestmentState` in `apply_invest_state_transition`:

https://github.com/centrifuge/centrifuge-chain/blob/0b6924df3d14d6c18a15f760746259baca6b47c2/pallets/foreign-investments/src/impls/mod.rs#L1076-L1083

Reply:
Thanks for the details!